vault/external_tests/audit/audit_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package audit
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/api/auth/userpass"
+	"github.com/hashicorp/vault/helper/testhelpers/minimal"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAudit_HMACFields verifies that all appropriate fields in audit
+// request and response entries are HMACed properly. The fields in question are:
+//   - request.headers.x-correlation-id
+//   - request.data: all sub-fields
+//   - all sub-fields
+//   - respnse.auth.client_token
+//   - response.auth.accessor
+//   - response.data: all sub-fields
+//   - response.wrap_info.token
+//   - response.wrap_info.accessor
+func TestAudit_HMACFields(t *testing.T) {
+	cluster := minimal.NewTestSoloCluster(t, nil)
+	client := cluster.Cores[0].Client
+
+	tempDir := t.TempDir()
+	logFile, err := os.CreateTemp(tempDir, "")
+	require.NoError(t, err)
+	devicePath := "file"
+	deviceData := map[string]any{
+		"type":        "file",
+		"description": "",
+		"local":       false,
+		"options": map[string]any{
+			"file_path": logFile.Name(),
+		},
+	}
+
+	_, err = client.Logical().Write("sys/config/auditing/request-headers/x-correlation-id", map[string]interface{}{
+		"hmac": true,
+	})
+	require.NoError(t, err)
+
+	// Request 1
+	_, err = client.Logical().Write("sys/audit/"+devicePath, deviceData)
+	require.NoError(t, err)
+
+	// Request 2
+	// Ensure the device has not been created.
```

Comment:
```suggestion
	// Ensure the device has been created.
```
(Assuming this will be right, based on the line 61 😄)

Reply:
🤦‍♂️ good catch